Testing RC-9 Provider Crash

Environment
Proxmox : 8.4.1
Provider: 3.0.1-rc9

Testing building 2 types of LCX (rocky9 and ubuntu 22.04)
Testing building 3 x VM's (Rocky8, Rocky9 and Ubuntu 220.4)
Configs updated from RC8 to RC9 to reflect CPU mapping changes


Test Environment builds correctly with terraform plan and apply
No changes made but a re-apply creates a plugin crash



 Error: Plugin did not respond
│
│   with proxmox_vm_qemu.lab-vm-rocky8,
│   on vm_rocky8.tf line 9, in resource "proxmox_vm_qemu" "lab-vm-rocky8":
│    9: resource "proxmox_vm_qemu" "lab-vm-rocky8" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The
│ plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with proxmox_vm_qemu.lab-vm-rocky9,
│   on vm_rocky9.tf line 2, in resource "proxmox_vm_qemu" "lab-vm-rocky9":
│    2: resource "proxmox_vm_qemu" "lab-vm-rocky9" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The
│ plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with proxmox_vm_qemu.lab-vm-ubuntu2204,
│   on vm_ubuntu.tf line 3, in resource "proxmox_vm_qemu" "lab-vm-ubuntu2204":
│    3: resource "proxmox_vm_qemu" "lab-vm-ubuntu2204" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The
│ plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v3.0.1-rc9 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 78 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.terraform_Disks_QemuDisks({0x0?, 0x0?, 0x0?, 0xc00072d300?}, 0x0?, 0x0?)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform_disks.go:37 +0x4f4
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk.Terraform_Unsafe(0xed5be8?, 0xc000d894e0, 0xc?)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/disk/terraform.go:13 +0xe8
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuRead({0xed5be8, 0xc0001f0e70}, 0xc00072d200, {0xc29e20?, 0xc0005b1130})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1178 +0x7aa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000488e00, {0xed5b40, 0xc00048be30}, 0xd?, {0xc29e20, 0xc0005b1130})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:823 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000488e00, {0xed5b40, 0xc00048be30}, 0xc00071f380, {0xc29e20, 0xc0005b1130})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:1117 +0x552
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0002ec5d0, {0xed5b40?, 0xc00048bd40?}, 0xc0001c7180)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:708 +0x6ca
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0001f28c0, {0xed5b40?, 0xc00048ae70?}, 0xc0001f0690)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:783 +0x309
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd65fa0?, 0xc0001f28c0}, {0xed5b40, 0xc00048ae70}, 0xc00072cf80, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:482 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000115200, {0xed5b40, 0xc00048ade0}, {0xed9e78, 0xc0001ea600}, 0xc0009e5d40, 0xc00048a8d0, 0x145af68, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000115200, {0xed9e78, 0xc0001ea600}, 0xc0009e5d40)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 17
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x135

Error: The terraform-provider-proxmox_v3.0.1-rc9 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


[lcx_rocky9.txt](https://github.com/user-attachments/files/20237389/lcx_rocky9.txt)
[lcx_ubuntu2204.txt](https://github.com/user-attachments/files/20237387/lcx_ubuntu2204.txt)
[vm_rocky8.txt](https://github.com/user-attachments/files/20237390/vm_rocky8.txt)
[vm_rocky9.txt](https://github.com/user-attachments/files/20237388/vm_rocky9.txt)
[vm_ubuntu.txt](https://github.com/user-attachments/files/20237391/vm_ubuntu.txt)

--- a/proxmox/Internal/resource/guest/qemu/disk/schema.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/schema.go
@@ -72,7 +72,7 @@ const (
 	slotScsi   string = schemaScsi
 	slotVirtIO string = schemaVirtIO
 
-	amountIdeSlots    = 3
+	amountIdeSlots    = 4
 	amountSataSlots   = 6
 	amountScsiSlots   = 31
 	amountVirtIOSlots = 16

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disks.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disks.go
@@ -284,7 +284,7 @@ func terraform_Disks_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *boo
 	return terraform_Disks_QemuCdRom(config.CdRom)
 }
 
-func terraform_Disks_QemuVirtIODisks(config *pveAPI.QemuVirtIODisks, schema any) any {
+func terraform_Disks_QemuVirtIODisks(config *pveAPI.QemuVirtIODisks, schema any) []any {
 	subSchemas := make([][]any, amountVirtIOSlots)
 	if v, ok := schema.([]any); ok && len(v) != 0 && v[0] != nil {
 		subSchema := v[0].(map[string]any)

--- a/proxmox/Internal/resource/guest/qemu/disk/terraform_disks.go
+++ b/proxmox/Internal/resource/guest/qemu/disk/terraform_disks.go
@@ -34,11 +34,11 @@ func terraform_Disks_QemuDisks(config pveAPI.QemuStorages, ciDisk *bool, d *sche
 	if v, ok := schema.([]any); ok && len(v) == 1 {
 		schemaMap = v[0].(map[string]any)
 	}
-	ide := terraform_Disks_QemuIdeDisks(config.Ide, ciDisk, schemaMap[schemaIDE].([]any))
-	sata := terraform_Disks_QemuSataDisks(config.Sata, ciDisk, schemaMap[schemaSata].([]any))
-	scsi := terraform_Disks_QemuScsiDisks(config.Scsi, ciDisk, schemaMap[schemaScsi].([]any))
-	virtio := terraform_Disks_QemuVirtIODisks(config.VirtIO, schemaMap[schemaVirtIO].([]any))
-	if ide != nil || sata != nil && scsi != nil && virtio != nil {
+	ide := terraform_Disks_QemuIdeDisks(config.Ide, ciDisk, schemaMap[schemaIDE])
+	sata := terraform_Disks_QemuSataDisks(config.Sata, ciDisk, schemaMap[schemaSata])
+	scsi := terraform_Disks_QemuScsiDisks(config.Scsi, ciDisk, schemaMap[schemaScsi])
+	virtio := terraform_Disks_QemuVirtIODisks(config.VirtIO, schemaMap[schemaVirtIO])
+	if ide != nil || sata != nil || scsi != nil || virtio != nil {
 		return []any{map[string]any{
 			schemaIDE:    ide,
 			schemaSata:   sata,
@@ -48,19 +48,22 @@ func terraform_Disks_QemuDisks(config pveAPI.QemuStorages, ciDisk *bool, d *sche
 	return nil
 }
 
-func terraform_Disks_QemuIdeDisks(config *pveAPI.QemuIdeDisks, ciDisk *bool, schema []any) []any {
-	var subSchema map[string]any
-	if len(schema) != 0 && schema[0] != nil {
-		subSchema = schema[0].(map[string]any)
+func terraform_Disks_QemuIdeDisks(config *pveAPI.QemuIdeDisks, ciDisk *bool, schema any) []any {
+	subSchemas := make([][]any, amountIdeSlots)
+	if v, ok := schema.([]any); ok && len(v) != 0 && v[0] != nil {
+		subSchema := v[0].(map[string]any)
+		for i := 0; i < amountIdeSlots; i++ {
+			subSchemas[i] = subSchema[schemaIDE+strconv.Itoa(i)].([]any)
+		}
 	}
 	if config == nil {
 		return nil
 	}
 	return []interface{}{map[string]interface{}{
-		schemaIDE + "0": terraform_Disks_QemuIdeStorage(config.Disk_0, ciDisk, subSchema[schemaIDE+"0"].([]any)),
-		schemaIDE + "1": terraform_Disks_QemuIdeStorage(config.Disk_1, ciDisk, subSchema[schemaIDE+"1"].([]any)),
-		schemaIDE + "2": terraform_Disks_QemuIdeStorage(config.Disk_2, ciDisk, subSchema[schemaIDE+"2"].([]any)),
-		schemaIDE + "3": terraform_Disks_QemuIdeStorage(config.Disk_3, ciDisk, subSchema[schemaIDE+"3"].([]any))}}
+		schemaIDE + "0": terraform_Disks_QemuIdeStorage(config.Disk_0, ciDisk, subSchemas[0]),
+		schemaIDE + "1": terraform_Disks_QemuIdeStorage(config.Disk_1, ciDisk, subSchemas[1]),
+		schemaIDE + "2": terraform_Disks_QemuIdeStorage(config.Disk_2, ciDisk, subSchemas[2]),
+		schemaIDE + "3": terraform_Disks_QemuIdeStorage(config.Disk_3, ciDisk, subSchemas[3])}}
 }
 
 func terraform_Disks_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool, schema []any) []any {
@@ -113,10 +116,10 @@ func terraform_Disks_QemuIdeStorage(config *pveAPI.QemuIdeStorage, ciDisk *bool,
 	return terraform_Disks_QemuCdRom(config.CdRom)
 }
 
-func terraform_Disks_QemuSataDisks(config *pveAPI.QemuSataDisks, ciDisk *bool, schema []any) []any {
+func terraform_Disks_QemuSataDisks(config *pveAPI.QemuSataDisks, ciDisk *bool, schema any) []any {
 	subSchemas := make([][]any, amountSataSlots)
-	if len(schema) != 0 && schema[0] != nil {
-		subSchema := schema[0].(map[string]any)
+	if v, ok := schema.([]any); ok && len(v) != 0 && v[0] != nil {
+		subSchema := v[0].(map[string]any)
 		for i := 0; i < amountSataSlots; i++ {
 			subSchemas[i] = subSchema[schemaSata+strconv.Itoa(i)].([]any)
 		}
@@ -183,10 +186,10 @@ func terraform_Disks_QemuSataStorage(config *pveAPI.QemuSataStorage, ciDisk *boo
 	return terraform_Disks_QemuCdRom(config.CdRom)
 }
 
-func terraform_Disks_QemuScsiDisks(config *pveAPI.QemuScsiDisks, ciDisk *bool, schema []any) []any {
+func terraform_Disks_QemuScsiDisks(config *pveAPI.QemuScsiDisks, ciDisk *bool, schema any) []any {
 	subSchemas := make([][]any, amountScsiSlots)
-	if len(schema) != 0 && schema[0] != nil {
-		subSchema := schema[0].(map[string]any)
+	if v, ok := schema.([]any); ok && len(v) != 0 && v[0] != nil {
+		subSchema := v[0].(map[string]any)
 		for i := 0; i < amountScsiSlots; i++ {
 			subSchemas[i] = subSchema[schemaScsi+strconv.Itoa(i)].([]any)
 		}
@@ -281,10 +284,10 @@ func terraform_Disks_QemuScsiStorage(config *pveAPI.QemuScsiStorage, ciDisk *boo
 	return terraform_Disks_QemuCdRom(config.CdRom)
 }
 
-func terraform_Disks_QemuVirtIODisks(config *pveAPI.QemuVirtIODisks, schema []any) []any {
+func terraform_Disks_QemuVirtIODisks(config *pveAPI.QemuVirtIODisks, schema any) any {
 	subSchemas := make([][]any, amountVirtIOSlots)
-	if len(schema) != 0 && schema[0] != nil {
-		subSchema := schema[0].(map[string]any)
+	if v, ok := schema.([]any); ok && len(v) != 0 && v[0] != nil {
+		subSchema := v[0].(map[string]any)
 		for i := 0; i < amountVirtIOSlots; i++ {
 			subSchemas[i] = subSchema[schemaVirtIO+strconv.Itoa(i)].([]any)
 		}

